src/main/java/org/apache/commons/pool2/PooledObject.java

```diff
@@ -33,6 +34,8 @@
  */
 public interface PooledObject<T> extends Comparable<PooledObject<T>> {
 
+    ReentrantLock lock = new ReentrantLock();
```

Comment:
A global variable? That sounds like anti-pattern. -1 unless you can explain what I am misunderstanding.

Reply:
This lock is to replace synchronized(PooledObject) with ReentrantLock in the GenericObjectPool,BaseGenericObjectPool and GenericKeyedObjectPool class.

Reply:
Hello @zenghu1chen 
A mutable global variable in an interface is an anti-pattern IMO, you've not addressed this issue in your comment.